### PR TITLE
Configurable staggered versioning

### PR DESCRIPTION
**Pull request by trmfreitas**
### Purpose

This change allows configuration of staggered versioning mode. Specifically the intervals.

### Testing

Testing was done to check if input boxes on gui was showing correct information and that interval configuration actually changed something.

### Screenshots

image

### Documentation

I am not sure if the GUI should be changed. The change I did is not very user friendly. For me, and perhaps most advanced uses of versioning feature, a configurable section in the XML is good enough.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
So I suggest you implement this as an external versioner, there is no reason for the average user to have access to this.

**Comment by trmfreitas** [comment]
So, you don't think it is usable to have this config in the xml only?

**Comment by AudriusButkevicius** [comment]
It could be there, but given it's the first time I've heard anyone interested in tweaking these, I'd just suggest to have an external versioner instead.

**Comment by calmh** [comment]
I'm inclined to agree; it's difficult enough to understand already without adding this level of customization.

**Comment by Ferroin** [comment]
I agree this should probably be config only, not in the UI, it's pretty cluttered, and I doubt many people will use it (on the grounds that the default intervals are pretty standard for things like this).  That said, If I had the ability to configure intervals like this, and ideally control the number of intervals and frequency of copies kept as well, I would probably be using the staggered versioning for most things.

**Comment by trmfreitas** [comment]
So, should I invest some time in checking that we can have this in config only and not have problems with GUI?

**Comment by AudriusButkevicius** [comment]
So I am actually against having this in the config too, as I don't think it would get much meddling. The config already has a lot of cruft, but most of the cruft is there to be able to switch off a feature, in this case there is nothing to switch off, and tweaks can be done by just extracting it into an external versioner.

**Comment by AudriusButkevicius** [comment]
Yet that's me, let's see what others have to say

**Comment by imsodin** [comment]
I agree this should not be in the folder edit modal, that would cause more confusion than it would be useful. I am hesitant to shut something down entirely just because there will be few users. Especially when the alternative is scripting, which is a huge barrier for many users. On the other hand I am skeptical that the kind of user that doesn't want to do scripting is interested in tweaking this parameters.

TLDR:  
I don't have any decisive new arguments to bring to the table. I slightly tend towards allowing it in the config on the basis that it won't do much harm.

**Comment by trmfreitas** [comment]
Ok I will experiment having only the config.xml option. 
Thank you.

**Comment by imsodin** [comment]
Not much to experiment there: Remove all the gui stuff and you are set. However at the moment nothing seems yet to be decided whether this will make it in or not (and overall tendency is still not).

**Comment by trmfreitas** [comment]
Yes, thanks.

Em 19/10/2017 7:04 p.m., "Simon Frei" <notifications@github.com> escreveu:

**Comment by AudriusButkevicius** [comment]
@calmh make the call

**Comment by calmh** [comment]
No. The staggered versioner is already too complicated; we should not make it worse.

**Comment by AudriusButkevicius** [comment]
Senior Techical Architect has spoken, amen.


Sorry about that @trmfreitas

**Comment by lkwg82** [comment]
Thx for your PR. Next time I encourage you to scribble and discuss the idea in the forum to avoid declined PR.

I went the same way ;).

**Comment by trmfreitas** [comment]
No problem. I did this cause I needed this kind of versioning to use
syncthing as a backup solution on one of my machines. With this I am able
to schedule backups to be kept for one week, one per day, one month, one
per week, and every month of the year. That is why I did it and then I
thought that it would be great if I didn't have to use my own compiled
version so I thought about checking if this was interesting for someone
else.

Thank you all for your time and a very nice product. 😀
